src/main/java/org/logstash/beats/BeatsHandler.java

```diff
@@ -0,0 +1,109 @@
+package org.logstash.beats;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelHandler;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+import io.netty.handler.timeout.IdleState;
+import io.netty.handler.timeout.IdleStateEvent;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.LogManager;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+@ChannelHandler.Sharable
+public class BeatsHandler extends ChannelInboundHandlerAdapter {
```

Comment:
if you use `SimpleChannelInboundHandler`, you can use generics and dont need to cast in `channelRead`